apps/pipelines/nodes/base.py

```diff
@@ -91,6 +109,7 @@ def process(self, node_id: str, incoming_edges: list, state: PipelineState, conf
                 break
         else:  # This is the first node in the graph
             input = state["messages"][-1]
+            state["shared_state"]["user_input"] = input
```

Comment:
Why not do this where the state is initially created?

Reply:
I think we'd have to remember to do it everywhere we call `invoke` on the pipeline, and this seemed like something the caller shouldn't have to care about.